pkg/wireguard/agent/agent.go

```diff
@@ -57,8 +61,9 @@ type Agent struct {
 	wgClient         *wgctrl.Client
 	listenPort       int
 	privKey          wgtypes.Key
-	pubKeyByNodeName map[string]string // nodeName => pubKey
-	restoredPubKeys  map[string]struct{}
+	peerByNodeName   map[string]*peerConfig
```

Comment:
Why don't we store the node IP address as a key of this map instead of having the `nodeNameByNodeIP` with redundant information?

Reply:
I considered it, but it's not better. A peer will have two IPs (v4+v6), but only ever a single name. This makes the name the better choice as a key.

I need to lookup by name for `UpdatePeer` and `DeletePeer` (to deal with IP change). This means I either need `nodeNameByIP` (which I have right now) or `nodeIPsByNodeName`, but the latter proved annoying because I suddenly had to keep two entries in the map. Having it indexed by name also allowed me to get rid of `pubKeyByNodeName`.